src/Geometry.cc

```diff
@@ -248,97 +204,97 @@ void Geometry::SetType(const GeometryType _type)
 /////////////////////////////////////////////////
 const Box *Geometry::BoxShape() const
```

Comment:
Could these getters now return a `const std::optional<T> &` or would that be too breaking of a change? It would probably save double checking if the optional type is valid. (once here in optionalToPointer and again at the calling site). It would just be pushed to the caller to check that the optional is valid.

Reply:
Even in this case, the caller has to check that the returned pointer is valid, so I don't think it's too much of a stretch

Reply:
I would love to do that, but I think that would break existing code without a tick-tock.

Reply:
If that's a desirable change, then it seems like it makes sense to start the tick-tock cycle. But how would that look if you're just changing the return type? Would you need to create new method name?